app/javascript/orangelight/availability.es6

```diff
@@ -194,7 +201,14 @@ export default class AvailabilityUpdater {
         const { label, cdl, temp_location } = holding_records[id][holding_id];
         // case :constituent with host ids.
         // data-record-id has a different this.id when there are host ids.
-        let availability_element = $(`*[data-availability-record='true'][data-record-id='${id}'][data-holding-id='${holding_id}'] .availability-icon`);
+        let availability_element;
+
+        // If we are not getting holding info select the availavility element by record id only.
```

Comment:
This comment can be misleading. We are getting holding info. We dont get an actual holding_id in the availability response. We get the temporary location code as a holding_id. But the rest is holding information. 
```
  "99125473625906421": {
    "RES_SHARE$IN_RS_REQ": {
      "on_reserve": "N",
      "location": "RES_SHARE$IN_RS_REQ",
      "label": "Resource Sharing Library - Lending Resource Sharing Requests",
      "status_label": "Unavailable",
      "copy_number": null,
      "cdl": false,
      "temp_location": true,
      "id": "RES_SHARE$IN_RS_REQ"
    }
  }
}
```
the `data-holding-id='${holding_id}' in the DOM is coming from https://catalog-qa.princeton.edu/catalog/99125473625906421/raw specifically the holdings_1display field. Then we do the match with the availability response. 
The specific location `RES_SHARE$IN_RS_REQ` even though it is a temporary location (has the location code as a holding id in the availability response), it is indexed as a permanent location (check the holdings_1display). This is why there is no direct much. 
To update the DOM we use the https://github.com/pulibrary/orangelight/blob/main/app/services/physical_holdings_markup_builder.rb . There is some code to handle the temporary location and set the holding_id for the request button so that the user can load the request form. https://github.com/pulibrary/orangelight/blob/70f77abb34bfe0401d7b04ba375daca8b8c35df3/app/services/physical_holdings_markup_builder.rb#L285

Maybe try to update the `holding_id` in the element with class holding-status , with the value of the temporary location code only when it is a `RES_SHARE$IN_RS_REQ`. I haven't tested this locally but let me know if you want me to take a look. https://github.com/pulibrary/orangelight/blob/70f77abb34bfe0401d7b04ba375daca8b8c35df3/app/services/physical_holdings_markup_builder.rb#L49

Reply:
I meant to paste this line with class: holding-status. https://github.com/pulibrary/orangelight/blob/70f77abb34bfe0401d7b04ba375daca8b8c35df3/app/services/physical_holdings_markup_builder.rb#L61